src/pkg/build/assemblers/assembler_sif.go

```diff
@@ -77,6 +79,14 @@ func createSIF(path string, definition []byte, squashfile string) (err error) {
 		return fmt.Errorf("while creating container: %s", err)
 	}
 
+	// chown the sif file to the calling user
+	if syscall.Getuid() == 0 {
```

Comment:
What about the case where the calling user is root. Not running `sudo ...` but they are logged in as the root user? In this case, `$SUDO_UID` may not be set, and since we're not checking for `Atoi` errors, we can't be guaranteed of the behavior